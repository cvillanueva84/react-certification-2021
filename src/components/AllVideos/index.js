export { default } from './AllVideos.component';