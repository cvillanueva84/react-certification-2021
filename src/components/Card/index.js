export {default} from './Card.component.jsx'; 