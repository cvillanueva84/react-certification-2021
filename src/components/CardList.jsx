import React from 'react';
import Grid from '@material-ui/core/Grid';
import VideoCard from './VideoCard';
import videos from '../youtube-videos-mock';

export default function CardList() {
  return (
    <Grid
      container
      justifyContent="center"
      alignItems="center"
      style={{ padding: '30px' }}
    >
      {videos.items.map((video) => (
        <VideoCard videoInfo={video.snippet} key={video.etag} />
      ))}
    </Grid>
  );
}
