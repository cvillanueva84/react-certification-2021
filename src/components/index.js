export { default as Navbar } from './Navbar/Navbar';
export { default as SearchBar } from './SearchBar/SearchBar';
export { default as ThemeSwitch } from './ThemeSwitch/ThemeSwitch';
export { default as Results } from './Results/Results';
