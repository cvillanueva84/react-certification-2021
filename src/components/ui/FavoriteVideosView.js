import React, { useContext } from 'react'
import { ContenedorStyled, Title } from '../../style/HomeScreenStyled';
import { GlobalContext } from '../Context'
import { FavItem } from '../FavItem';
import { HeaderNav } from '../HeaderNav';
import { HeaderFav, CardFavs } from '../../style/FavoriteVideosViewStyled';


export const FavoriteVideosView = () => {

    const { myStateReducer } = useContext(GlobalContext);

    return (
        <>
            <HeaderFav>
                <HeaderNav />
            </HeaderFav>

            <ContenedorStyled>
                <Title>Favorite Videos:</Title>
                <CardFavs>
                    {
                        myStateReducer.favorites.map(vid => (
                            <FavItem
                                key={vid.videoId}
                                {...vid}
                            />
                        ))
                    }
                </CardFavs>
            </ContenedorStyled>
        </>
    )
}
