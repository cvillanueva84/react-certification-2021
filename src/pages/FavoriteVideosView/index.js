export { default } from './FavoriteVideosView.page';
