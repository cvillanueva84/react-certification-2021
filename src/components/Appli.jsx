import React from 'react';
import CardList from './CardList';
import NavBar from './NavBar';

export default function Appli() {
  return (
    <div>
      <NavBar />
      <CardList />
    </div>
  );
}
