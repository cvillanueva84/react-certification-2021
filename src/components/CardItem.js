import React from 'react';
import { useContext } from 'react';
import { GlobalContext } from './Context';
import {CardItemUi}  from './ui/CardItemUi'

export const CardItem = ({ title, description, url, videoId }) => {

  const { myStateReducer, dispatch } = useContext(GlobalContext);

  const handleClick = () => {
      dispatch({
        type: 'actionAddVideoData',
        payload: {
          ...myStateReducer,
          videoData: {
            title: title,
            description: description,
            url: url,
            videoId: videoId
          }
        }
      });
  }

  return (
    <CardItemUi 
      title={title} 
      description={description} 
      url={url} 
      handleClick={handleClick}
    />
  );

};

