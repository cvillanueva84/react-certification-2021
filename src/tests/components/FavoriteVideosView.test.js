import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { GlobalContext } from '../../components/Context';
import { MemoryRouter, Router } from 'react-router';
import { FavoriteVideosView } from "../../components/ui/FavoriteVideosView";


describe('Test Favorite Videos View', () => {
    const contextValue = {
        dispatch: jest.fn(),
        myStateReducer: {
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }
    }
    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const wrapper = mount(
        <GlobalContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <FavoriteVideosView />
                </Router>
            </MemoryRouter>
        </GlobalContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render success', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists());
    });
});