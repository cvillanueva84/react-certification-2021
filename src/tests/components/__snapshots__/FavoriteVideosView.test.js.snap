// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Favorite Videos View should render success 1`] = `ReactWrapper {}`;
